src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -0,0 +1,210 @@
+package org.sunbird.profile.service;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.sunbird.common.model.SBApiResponse;
+import org.sunbird.common.util.CbExtServerProperties;
+import org.sunbird.common.util.Constants;
+import org.sunbird.core.logger.CbExtLogger;
+
+import java.util.*;
+
+@Component
+@Service
+public class ProfileServiceImpl implements ProfileService{
```

Comment:
Why this class is annotated with both Component and Service ?
Why are we are still using constant strings directly -- like "update", "transition" etc ? this suppose to be from Constants class? 
Need to be more careful whenever we create new object. Need to check how to avoid using temp variables and be productive -- otherwise it would be an performance issue.